library/network_connections.py

```diff
@@ -676,7 +676,11 @@ def setting_ip_config_get_routes(self, s_ip):
     def connection_ensure_setting(self, connection, setting_type):
         setting = connection.get_setting(setting_type)
         if not setting:
-            setting = setting_type.new()
+            if setting_type is Util.NM().SettingMatch:
```

Comment:
Since you are not guaranteed to be using a version of the nm library that has `SettingMatch`, there are a couple of ways to do this - I suggest doing the following:
```python
             util_nm = Util.NM()
             if hasattr(util_nm, 'SettingMatch') and setting_type is util_nm.SettingMatch:
               setting = util_nm.SettingMatch()
             else:
               setting = setting_type.new()
```
That is - ensure that the module has the `SettingMatch` attribute first before trying to use it.
but I'll defer to @thom311 @cathay4t if they have a better approach to handle this